src/plugins/saved_objects_management/public/management_section/objects_table/components/table.test.tsx

```diff
@@ -124,9 +124,33 @@ describe('Table', () => {
       selectedSavedObjects,
       capabilities: { savedObjectsManagement: { delete: false } } as any,
     };
-    const component = shallowWithI18nProvider(<Table {...customizedProps} />);
+    render(
+      <I18nProvider>
+        <Table {...customizedProps} />
+      </I18nProvider>
+    );
 
-    expect(component).toMatchSnapshot();
+    await waitFor(() => {
+      expect(screen.getByTestId('savedObjectsManagementDelete')).toBeDisabled();
+    });
+  });
+
+  it(`disables delete when no objects are selected `, async () => {
+    const selectedSavedObjects = [] as any;
+    const customizedProps = {
+      ...defaultProps,
+      selectedSavedObjects,
+      capabilities: { savedObjectsManagement: { delete: true } } as any,
+    };
+    render(
+      <I18nProvider>
+        <Table {...customizedProps} />
+      </I18nProvider>
+    );
+
+    await waitFor(() => {
+      expect(screen.getByTestId('savedObjectsManagementDelete')).toBeDisabled();
```

Comment:
Assert the button's disabled rather than relying on a snapshot, changes to which are easily overlooked (or "fixed" without inspection)